evap/grades/templates/grades_semester_view.html

```diff
@@ -97,6 +99,40 @@ <h3 class="panel-title">{% trans "Courses" %}</h3>
     </div>
 {% endblock %}
 
+{% block modals %}
+    {{ block.super }}
+    {% trans "Have final grades been submitted?" as title %}
+    {% trans 'Please confirm that the final grades for the course <strong id="data-label"></strong> have been submitted but will not be uploaded.' as question %}
+    {% trans "Confirm" as action_text %}
+    {% include_confirmation_modal "confirm_noupload_modal" title question action_text "primary" %}
+    <script type="text/javascript">
+        function confirm_noupload_modal_action(data_id) {
+            $.ajax({
+                type: "POST",
+                url: "{% url "grades:toggle_no_grades" %}",
+                data: {"course_id": data_id},
+                success: function(){ location.reload(); },
+                error: function(){}
+            });
+        };
+    </script>
+    {% trans "Will final grades be uploaded?" as title %}
+    {% trans 'Please confirm that a grade document for the course <strong id="data-label"></strong> will be uploaded later on.' as question %}
+    {% trans "Confirm" as action_text %}
+    {% include_confirmation_modal "confirm_laterupload_modal" title question action_text "primary" %}
+    <script type="text/javascript">
+        function confirm_laterupload_modal_action(data_id) {
+            $.ajax({
+                type: "POST",
+                url: "{% url "grades:toggle_no_grades" %}",
+                data: {"course_id": data_id},
+                success: function(){ location.reload(); },
+                error: function(){}
+            });
+        };
+    </script>
+{% endblock %}
```

Comment:
although possible i won't implement it. the reload will stay, otherwise every change in the page layout would need changes in the javascript as well and there are too many elements involved which have to be changed so there is a high chance of creating inconsistencies.